packages/vow/src/types.ts

```diff
@@ -7,6 +7,16 @@ import type { CopyTagged, RemotableObject } from '@endo/pass-style';
  */
 export type IsRetryableReason = (reason: any, priorRetryValue: any) => any;
 
+/**
+ * Types that are conceptually similar to Promise<F>.
+ */
+export type Promissory<F> = Promise<F> | PromiseLike<F> | Vow<F>; // s| PromiseStep<F> …etc.
```

Comment:
woah a new concept. how about `VowLike`?

Reply:
`VowLike` it is!